sources/Application/Views/InstrumentView.cpp

```diff
@@ -939,6 +952,22 @@ void InstrumentView::Update(Observable &o, I_ObservableData *data) {
     SetChanged();
     NotifyObservers(&ve);
   } break;
+  case FourCC::SampleInstrumentSlices: {
+    // We are assuming we can only get here when instrument is Sample, safe?
+    Trace::Debug("INSTRUMENTVIEW", "slice changed, redraw!");
+    I_Instrument *instrument = getInstrument();
+    // In slice change, reset markers for start and loop start
+    Variable *ls = instrument->FindVariable(FourCC::SampleInstrumentStart);
+    ls->SetInt(0, true);
+    ls = instrument->FindVariable(FourCC::SampleInstrumentLoopStart);
+    ls->SetInt(0, true);
+    Variable *slices = instrument->FindVariable(FourCC::SampleInstrumentSlices);
+    ls = instrument->FindVariable(FourCC::SampleInstrumentEnd);
+    ls->SetInt((((SampleInstrument *)instrument)->GetSampleSize() - 1) /
+                   slices->GetInt(),
+               true);
+    refreshInstrumentFields(instrument, FourCC::SampleInstrumentSlices);
```

Comment:
We'll want to create a more targeted means to update the sample related fields here as refreshInstrumentFields does far too much for this and may even cause a crash I think being called from here.

Reply:
Actually no, it happens when you come into instruments AND if you change the instrument type.
But thats my whole point, its a change across an entire instrument **type** that its intended for, its not meant for just updating a couple of fields.
But all good, I already put up my hand to make a lighter and simpler way to just update the 3 required sample start/end fields 👍🏻